tests/contrib/openai/test_logger.py

```diff
@@ -0,0 +1,142 @@
+import os
+import time
+
+import mock
+import pytest
+import vcr
+
+from ddtrace.contrib.openai._logging import V2LogWriter
+from tests.utils import request_token
+
+
+logs_vcr = vcr.VCR(
+    cassette_library_dir=os.path.join(os.path.dirname(__file__), "cassettes/"),
+    record_mode="once",
+    match_on=["path"],
+    filter_headers=[("DD-API-KEY", "XXXXXX")],
+    # Ignore requests to the agent
+    ignore_localhost=True,
+)
+
+
+@pytest.fixture(autouse=True)
+def vcr_logs(request):
+    marks = [m for m in request.node.iter_markers(name="vcr_logs")]
+    assert len(marks) < 2
+    if marks:
+        mark = marks[0]
+        cass = mark.kwargs.get("cassette", request_token(request).replace(" ", "_").replace(os.path.sep, "_"))
+        with logs_vcr.use_cassette("%s.yaml" % cass):
+            yield
+    else:
+        yield
+
+
+@pytest.fixture
+def mock_logs():
```

Comment:
this is better than `caplog` because using `.assert_has_calls` means we don't need to worry about log formatter/timestamps/etc ?